### DISCUSSION TITLE
[BETA] v2 with icon picker and lazy loading support

### DISCUSSION THREAD

**Discussion by vigonotion** [vigonotion/hass-simpleicons#194] [Announcements]
I took some time and implemented lazy loading and the new icon picker added in HA 2021.11. As this is a completely new way of showing the icons, I'd like to gather some feedback if it works for you.

![grafik](https://user-images.githubusercontent.com/1506970/140404259-38c813f9-ab2b-45af-aa3d-4b4c54d12eaf.png)

You can help me by installing the beta version in HACS: Go to HACS > Integrations > simpleicons and click on the three dots, then reinstall and select show beta versions and install the newest `v2` beta.
Add a comment below if it works for you or if not, what HA version you are using, what browser etc.

Check out the release here: https://github.com/vigonotion/hass-simpleicons/releases/tag/v2.0.0-rc.1

**Comment by nepozs** [comment]
Firefox 93.0 windows x64, network LAN and VPN (both connections tested), ~~simpleicons works (as usual).~~
but

Icon picker does NOT work with any si: icon, but when full name is typed icon works.
![simpleicons_1](https://user-images.githubusercontent.com/56661567/140443510-00331e27-5b63-41c1-a40d-85ebb27f7804.png)
![simpleicons_2](https://user-images.githubusercontent.com/56661567/140443520-2c7dd6ed-e02c-42b0-89d4-23b1a8fa98d8.png)
![simpleicons_3](https://user-images.githubusercontent.com/56661567/140443578-7be5efc4-5c84-4f7f-81e7-f2a8176b7dc1.png)

## System Health

version | core-2021.11.0
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.9.7
os_name | Linux
os_version | 5.10.75
arch | x86_64
timezone | Europe/Warsaw

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
Github API Calls Remaining | 5000
Installed Version | 1.16.0
Stage | running
Available Repositories | 892
Installed Repositories | 54

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>GIOŚ</summary>

can_reach_server | ok
-- | --

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 6.6
-- | --
update_channel | stable
supervisor_version | supervisor-2021.10.8
docker_version | 20.10.8
disk_total | 14.0 GB
disk_used | 11.1 GB
healthy | true
supported | true
board | generic-x86-64
supervisor_api | ok
version_api | ok
installed_addons | File editor (5.3.3), SSH & Web Terminal (9.0.1), Samba share (9.5.1), ZeroTier One (0.11.2), FTP (4.2.1), Glances (0.13.0), Check Home Assistant configuration (3.9.0), Local VLC (18), Mosquitto broker (6.0.1), Zigbee2mqtt (1.22.0-2), Tasmota Device Manager (2021.7.0), HDD Tools (0.51.0), motionEye (0.15.1)

</details>

<details><summary>Lovelace</summary>

dashboards | 2
-- | --
resources | 26
views | 23
mode | storage

</details>

<details><summary>Spotify</summary>

api_endpoint_reachable | ok
-- | --

</details>

<details><summary>Żadnego Ale</summary>

can_reach_server | ok
-- | --

</details>

Edit:
After update firefox to 94.0.1 icon picker works
![simpleicons_4](https://user-images.githubusercontent.com/56661567/140446978-f7654d53-57af-4976-9747-a637e61b9ea7.png)

BUT

there is serious error with header - after click "Poziomo" it is name of non-default dashboard
![simpleicons_5_error](https://user-images.githubusercontent.com/56661567/140447697-03ed07a1-f804-4355-b27e-7f6366d52ac6.png)

this part should look like
![simpleicons_6_after_reload](https://user-images.githubusercontent.com/56661567/140447740-a03d7e90-0ebd-4ec6-a65f-8b4957639f96.png)

and there is one more with left sidebar
![simpleicons_7](https://user-images.githubusercontent.com/56661567/140448638-8a15eb77-93cc-44bf-9a42-084fc11f1268.png)
after change default dashboard
![simpleicons_8](https://user-images.githubusercontent.com/56661567/140448641-1c65dd9b-64be-46cc-8993-10c3ff0bee13.png)

**Reply by vigonotion** [reply]
I think it was not the firefox update but the update reloaded ha without cache. Nevertheless, the sidebar icons getting too big is so strange, because those aren't even simpleicons

**Reply by nepozs** [reply]
I've cleared cache before, so it can be related to ff version also, but now it is even stranger - I've reverted to stable v1.20.0 and it does not help…

I'm using simpleicons also with my 2 production installations, and they work properly (this one also has worked OK before installing beta).

**Reply by nepozs** [reply]
Update - it is conflict of (beta version?) simpleicons with fontawesome (it was coexisting with simpleicons with no errors very long time #35 and I forget about fontawesome, as I don't use it in real installations)

**Reply by ildar170975** [reply]
> Icon picker does NOT work with any si: icon, but when full name is typed icon works.
Same.
v2.0.0-rc.1
2021.11.3
Win10x64 + Chrome  95.0.4638.69 + Firefox 94

At least now I see icons manually typed in the Editor!

**Comment by drinfernoo** [comment]
Working great here with the latest update 👍
